client/src/main/java/io/pravega/client/auth/CredentialsHelper.java

```diff
@@ -0,0 +1,111 @@
+    private static final String AUTH_METHOD_LOAD_DYNAMIC = AUTH_PROPS_START + "loadDynamic";
+
+    private static final String AUTH_PROPS_START_ENV = "pravega_client_auth_";
+
+    /**
+     * Function to extract the credentials object in the given client config.
+     * Here is the order of preference in descending order:
+     * 1. User provides a credential object. This overrides any other settings.
+     * 2. System properties: System properties are defined in the format: "pravega.client.auth.*"
+     * 3. Environment variables. Environment variables are defined under the format "PRAVEGA_CLIENT_AUTH_*"
+     * 4. In case of option 2 and 3, the caller can decide whether the class needs to be loaded dynamically by
+     *     setting property `pravega.client.auth.loadDynamic` to true.
+     *
+     * @param config the config containing the credentials object.
+     */
+    public static ClientConfig extractCredentials(ClientConfig config) {
+        if (config.getCredentials() != null) {
+            return config;
+        }
+
+        Credentials credentials = extractCredentialsFromProperties();
+        if (credentials == null) {
+            credentials = extractCredentialsFromEnv();
+        }
+
+        return ClientConfig.builder()
+                           .credentials(credentials)
+                           .trustStore(config.getTrustStore())
+                           .validateHostName(config.isValidateHostName())
+                           .controllerURI(config.getControllerURI())
+                           .build();
+
+    }
+
+    private static Credentials extractCredentialsFromProperties() {
+        Map<String, String> retVal = System.getProperties().entrySet()
+                                           .stream()
+                                           .filter(entry -> entry.getKey().toString().startsWith(AUTH_PROPS_START))
+                                           .collect(Collectors.toMap(entry ->
+                                                   entry.getKey().toString().replace("_", "."),
+                                                   value -> (String) value.getValue()));
+        if (retVal.containsKey(AUTH_METHOD)) {
+            return credentialFromMap(retVal);
+        } else {
+            return null;
+        }
+    }
+
+    private static Credentials extractCredentialsFromEnv() {
+        Map<String, String> retVal = System.getenv().entrySet()
+                                           .stream()
+                                           .filter(entry -> entry.getKey().toString().startsWith(AUTH_PROPS_START_ENV))
+                                           .collect(Collectors.toMap(entry -> (String) entry.getKey(), value -> (String) value.getValue()));
+        if (retVal.containsKey(AUTH_METHOD)) {
+            return credentialFromMap(retVal);
+        } else {
+            return null;
+        }
+    }
+
+    private static Credentials credentialFromMap(Map<String, String> retVal) {
```

Comment:
Find a better name for this parameter. It's not a return value

Reply:
Fixed.

Reply:
Fixed.